pkg/validation/validation.go

```diff
@@ -85,6 +89,20 @@ func nodeGroupPools(nodeGroups []nropv1.NodeGroup) error {
 	return nil
 }
 
+func nodeGroupsValidPoolName(nodeGroups []nropv1.NodeGroup) error {
+	var err error
+	for idx, nodeGroup := range nodeGroups {
+		if nodeGroup.PoolName == nil {
+			continue
+		}
+		if *nodeGroup.PoolName != "" {
```

Comment:
Uhm, the spaces so? thing is a name composed only by empty spaces (like "    ") is... legal and our code should be able to handle it. If an user really wants such a name... so be it.
The reason why empty string is rejected is because it breaks some internal assumptions (corner case) but past that, we don't really care about the content as long as it is unique (and, indeed, not-empty)